com.unity.shadergraph/Editor/Drawing/Inspector/PropertyDrawers/ShaderInputPropertyDrawer.cs

```diff
@@ -1450,10 +1457,203 @@ string GetSanitizedReferenceName(string name)
             return Regex.Replace(name, m_ReferenceNameDisallowedPattern, "_");
         }
 
-        int FindDuplicateReferenceNameIndex(int id, string referenceName)
+        int FindDuplicateKeywordReferenceNameIndex(int id, string referenceName)
         {
             var entryList = m_KeywordReorderableList.list as List<KeywordEntry>;
             return entryList.FindIndex(entry => entry.id != id && entry.referenceName == referenceName);
         }
+
+        void BuildDropdownFields(PropertySheet propertySheet, ShaderInput shaderInput)
+        {
+            var dropdown = shaderInput as ShaderDropdown;
+            if (dropdown == null)
+                return;
+
+            BuildDropdownField(propertySheet, dropdown);
+            BuildExposedField(propertySheet);
+        }
+
+        void BuildDropdownField(PropertySheet propertySheet, ShaderDropdown dropdown)
+        {
+            // Clamp value between entry list
+            int value = Mathf.Clamp(dropdown.value, 0, dropdown.entries.Count - 1);
+
+            // Default field
+            var field = new PopupField<string>(dropdown.entries.Select(x => x.displayName).ToList(), value);
+            field.RegisterValueChangedCallback(evt =>
+            {
+                this._preChangeValueCallback("Change Dropdown Value");
+                dropdown.value = field.index;
+                this._postChangeValueCallback(false, ModificationScope.Graph);
+            });
+
+            AddPropertyRowToSheet(propertySheet, field, "Default");
+
+            var container = new IMGUIContainer(() => OnDropdownGUIHandler()) { name = "ListContainer" };
+            AddPropertyRowToSheet(propertySheet, container, "Entries");
+            container.SetEnabled(true);
+        }
+
+        void OnDropdownGUIHandler()
```

Comment:
![image](https://user-images.githubusercontent.com/13033542/115779146-73d3fc00-a36c-11eb-8b4c-bc61ffe617b8.png)

The name of the re-orderable list seems to be uninitialized for some reason 🤔

Reply:
Are you referring to the "IList"?

Reply:
I fixed this, added the header, and it now says "Entry Name" instead of "IList".

Reply:
Yup that was it